afterpay/src/main/kotlin/com/afterpay/android/Afterpay.kt

```diff
@@ -258,12 +268,127 @@ object Afterpay {
+            items = items,
+            configuration = configuration,
+            isCashAppPay = true,
+        )
+
+        val checkoutResponseResult = runCatching {
+            val checkoutUrl = configuration.v3CheckoutUrl
+            val checkoutPayload = requireNotNull(Json.encodeToString(checkoutRequest))
+            return@runCatching withContext(Dispatchers.IO) {
+                ApiV3.request<CheckoutV3.Response, String>(
+                    checkoutUrl,
+                    ApiV3.HttpVerb.POST,
+                    checkoutPayload,
+                )
+            }.getOrThrow()
+        }
+
+        // requesting checkout data failed
+        checkoutResponseResult.onFailure { error: Throwable ->
+            return failure(error)
+        }
+
+        // requesting checkout data success: sign the checkout response token
+        checkoutResponseResult.onSuccess { result: CheckoutV3.Response ->
+            AfterpayCashAppCheckout.performSignPaymentRequest(result.token)
+                .let { cashAppSignOrderResult: CashAppSignOrderResult ->
+                    return when (cashAppSignOrderResult) {
+                        // signing failed
+                        is Failure -> failure(cashAppSignOrderResult.error)
+
+                        // signing success
+                        is Success -> {
+                            // map Result<CashAppSignOrderResult> to Result<CheckoutV3CashAppPay>
+                            success(
+                                CheckoutV3CashAppPay(
+                                    token = result.token,
+                                    singleUseCardToken = result.singleUseCardToken,
+                                    amount = cashAppSignOrderResult.response.amount,
+                                    redirectUri = cashAppSignOrderResult.response.redirectUri,
+                                    merchantId = cashAppSignOrderResult.response.merchantId,
+                                    brandId = cashAppSignOrderResult.response.brandId,
+                                    jwt = cashAppSignOrderResult.response.jwt,
+                                ),
+                            )
+                        }
+                    }
+                }
+        }
+
+        // should never happen, compiler doesn't know success and failure are only options
+        throw IllegalStateException()
+    }
+
+    /**
+     * Confirm that checkout was completed with Cash App Pay SDK
+     */
+    suspend fun confirmCheckoutV3WithCashAppPay(
+        customerId: String,
+        grantId: String,
+        token: String,
+        singleUseCardToken: String,
+        jwt: String,
+        configuration: CheckoutV3Configuration? = checkoutV3Configuration,
+    ): Result<CheckoutV3Data> {
+        return runCatching {
+            requireNotNull(configuration) {
+                "`configuration` must be set via `setCheckoutV3Configuration` or passed into this function"
+            }
```

Comment:
Just to make sure, due the `runCatching`, if this fails a `Result.Failure` will catch and encapsulate it right ?

Reply:
No. This will actually crash  since it happens before (outside) the `runCatching`. IMO this makes sense since this error would be caused by a developer error (not something inevitable like a network error)

Reply:
So something we did on PayKit, and putting it out there for consideration, was to soft crash instead of always crashing. That is, to crash if running in DEBUG, but to log an error and mutate to an error state if running in prod. The idea was to reduce the likelihood of our SDK being the thing causing a crash in production for merchants who did not do a controlled rollout.